server/application-server/src/main/java/de/tum/in/www1/hephaestus/codereview/pullrequest/Pullrequest.java

```diff
@@ -0,0 +1,110 @@
+package de.tum.in.www1.hephaestus.codereview.pullrequest;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.lang.NonNull;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.hephaestus.codereview.comment.IssueComment;
+import de.tum.in.www1.hephaestus.codereview.repository.Repository;
+import de.tum.in.www1.hephaestus.codereview.user.GHUser;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Table(name = "pull_request")
+@Getter
+@Setter
+@NoArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+@ToString
+public class PullRequest {
```

Comment:
I just saw that in the RestAPI `GHPullRequest` inherits from `GHIssue`. Maybe we can also do some inheritance. 

I also saw this article about it: https://www.baeldung.com/hibernate-inheritance

Maybe we can have a `@MappedSuperclass` `BaseGitServiceEntity` that we can extend everywhere adding the `id`, `created_at`, and `updated_at` there and in the future also `provider_id`.